pychunkedgraph/app/segmentation/common.py

```diff
@@ -424,40 +486,32 @@ def handle_split(table_id):
     # Call ChunkedGraph
     cg = app_utils.get_cg(table_id)
 
-    data_dict = {}
-    for k in ["sources", "sinks"]:
-        data_dict[k] = collections.defaultdict(list)
+    node_idents = []
+    coords = []
+    node_ids = []
 
+    for k in ["sources", "sinks"]:
         for node in data[k]:
-            node_id = node[0]
-            x, y, z = node[1:]
-            coordinate = np.array([x, y, z]) / cg.segmentation_resolution
-
-            atomic_id = cg.get_atomic_id_from_coord(
-                coordinate[0],
-                coordinate[1],
-                coordinate[2],
-                parent_id=np.uint64(node_id),
-            )
-
-            if atomic_id is None:
-                raise cg_exceptions.BadRequest(
-                    f"Could not determine supervoxel ID for coordinates "
-                    f"{coordinate}."
-                )
+            node_ids.append(node[0])
+            coords.append(np.array(node[1:]) / cg.segmentation_resolution)
+            node_idents.append(k)
 
-            data_dict[k]["id"].append(atomic_id)
-            data_dict[k]["coord"].append(coordinate)
+    node_ids = np.array(node_ids, dtype=np.uint64)
+    coords = np.array(coords)
+    node_idents = np.array(node_idents)
+    sv_ids = handle_supervoxel_id_lookup(cg, coords, node_ids)
 
-    current_app.logger.debug(data_dict)
+    current_app.logger.debug(
+        {"node_id": node_ids, "sv_id": sv_ids, "node_ident": node_idents}
+    )
 
     try:
         ret = cg.remove_edges(
             user_id=user_id,
-            source_ids=data_dict["sources"]["id"],
-            sink_ids=data_dict["sinks"]["id"],
-            source_coords=data_dict["sources"]["coord"],
-            sink_coords=data_dict["sinks"]["coord"],
+            source_ids=sv_ids[node_idents == "sources"],
```

Comment:
I think using integers here will be faster, string comparison will degrade if the array is big.
Maybe use `0` for `sources` and `1` for `sinks`?